dev/com.ibm.ws.beanvalidation.v20.cdi/src/com/ibm/ws/beanvalidation/v20/cdi/internal/ValidationReleasableFactoryImpl.java

```diff
@@ -263,34 +240,51 @@ private void createManagedConstraintValidatorFactory(Configuration<?> config, Cl
-//                                                                                         ValueExtractor.class
-//                        ) );
-//
-//        for ( ValueExtractor<?> valueExtractor : valueExtractors ) {
-//            valueExtractorDescriptors.add( new ValueExtractorDescriptor( injectInstance( valueExtractor ) ) );
-//        }
-//
-//        return valueExtractorDescriptors;
-//    }
+    private void addValueExtractorBeans(Configuration<?> config, ClassLoader appClassLoader) {
+        Map<ValueExtractorDescriptor.Key, ValueExtractorDescriptor> valueExtractorDescriptors = createValidationXmlValueExtractors(config,
+                                                                                                                                   appClassLoader).stream().collect(Collectors.toMap(ValueExtractorDescriptor::getKey,
+                                                                                                                                                                                     Function.identity()));
+
+        for (ValueExtractorDescriptor serviceLoaderValueExtractorDescriptor : createServiceLoaderValueExtractors()) {
+            valueExtractorDescriptors.putIfAbsent(serviceLoaderValueExtractorDescriptor.getKey(), serviceLoaderValueExtractorDescriptor);
+        }
+
+        for (ValueExtractorDescriptor valueExtractorDescriptor : valueExtractorDescriptors.values()) {
+            config.addValueExtractor(valueExtractorDescriptor.getValueExtractor());
+        }
+    }
+
+    private Set<ValueExtractorDescriptor> createValidationXmlValueExtractors(Configuration<?> config, ClassLoader appClassLoader) {
+        BootstrapConfiguration bootstrapConfiguration = config.getBootstrapConfiguration();
+        Set<String> valueExtractorClassNames = bootstrapConfiguration.getValueExtractorClassNames();
+
+        @SuppressWarnings("unchecked")
+        Set<ValueExtractorDescriptor> valueExtractorDescriptors = valueExtractorClassNames.stream() //
+                        .map(fqcn -> createManagedObject((Class<? extends ValueExtractor<?>>) loadClass(fqcn, appClassLoader))) //
+                        .map(valueExtractor -> new ValueExtractorDescriptor(valueExtractor)) //
+                        .collect(Collectors.toSet());
+
+        return valueExtractorDescriptors;
+    }
+
+    @SuppressWarnings("rawtypes")
+    private Set<ValueExtractorDescriptor> createServiceLoaderValueExtractors() {
+        Set<ValueExtractorDescriptor> valueExtractorDescriptors = new HashSet<>();
+
+        List<ValueExtractor> valueExtractors;
+
+        if (System.getSecurityManager() != null) {
+            valueExtractors = AccessController.doPrivileged(GetInstancesFromServiceLoader.action(PrivHelper.getContextClassLoader(), ValueExtractor.class));
+        } else {
+            valueExtractors = GetInstancesFromServiceLoader.action(PrivHelper.getContextClassLoader(), ValueExtractor.class).run();
```

Comment:
Why are we using PrivHelper on a path where System.getSecurityManager is null?

Reply:
Removed PrivHelper usage.